packages/gatsby/src/bootstrap/load-plugins/validate.ts

```diff
@@ -213,7 +213,8 @@ async function validatePluginsOptions(
       let gatsbyNode
       try {
         const resolvedPlugin = resolvePlugin(plugin, rootDir)
-        gatsbyNode = require(`${resolvedPlugin.resolve}/gatsby-node`)
+        gatsbyNode = require(resolvedPlugin.resolvedCompiledGatsbyNode ??
+          `${resolvedPlugin.resolve}/gatsby-node`)
```

Comment:
while not exactly part of the problem this PR is fixing currently, I wonder wether we should already do similar as you did in https://github.com/gatsbyjs/gatsby/pull/37440 to not use `require` for the plugins and instead use `await import` (or maybe we can use `importGatsbyPlugin` helper already at that point?)

Reply:
Yeah, I think I can just switch this out with the `importGatsbyPlugin` helper 👍